ddht/v5_1/alexandria/resource_queue.py

```diff
@@ -0,0 +1,56 @@
+from typing import Any, AsyncIterator, Collection, Container, Hashable, Set, TypeVar
+
+from async_generator import asynccontextmanager
+import trio
+
+TResource = TypeVar("TResource", bound=Hashable)
+
+
+class RemoveResource(Exception):
+    pass
+
+
+class ResourceQueue(Container[TResource]):
+    resources: Set[TResource]
+
+    def __init__(
+        self, resources: Collection[TResource], max_resource_count: int = 256
+    ) -> None:
```

Comment:
I think there's a bug if `resources` has more than 256 elements and the user does not provide a `max_resource_count`, if nothing is passed in it might want to default to something like `len(resources) + 256`.

This interface would be much easier to implement if we could resize memory channels. It looks like we *can*, by changing `self._send._state.max_buffer_size`, but I don't know how you feel about hard-coding trio implementation details.

I don't know where you plan on using this class so I don't know if this suggestion makes sense, but the easiest way to remove this class of bugs is to not have an `add(resource)` method.

Reply:
It's used here: 

https://github.com/ethereum/ddht/pull/95/files#diff-e8d0d620a600d82775b4794f8378624f90f1a7c0e9ed3e692da2a39e2a10de89R432-R503

The `add(resource)` is needed so that when we request a section of the content, and the response doesn't contain all of the requested data, we can add the sub-sections that are still missing back into the queue.

For now I'm not really intending to use this beyond that location.  I'll figure out a patch for the bug.